db/internal_stats.cc

```diff
@@ -2197,4 +2213,41 @@ std::unique_ptr<IntPropertyAggregator> CreateIntPropertyAggregator(
   }
 }
 
+void DBStatsCallback::OnSubcompactionBegin(const Compaction* c) {
+  if (c == nullptr) {
+    return;
+  }
+  TEST_SYNC_POINT("DBStatsCallback::OnSubcompactionBegin");
+  size_t runs_to_add = GetNumberCompactionSortedRuns(c);
+  num_running_compaction_sorted_runs_->fetch_add(runs_to_add,
+                                                 std::memory_order_relaxed);
+}
+
+void DBStatsCallback::OnSubcompactionEnd(const Compaction* c) {
+  if (c == nullptr) {
+    return;
+  }
+  size_t runs_finished = GetNumberCompactionSortedRuns(c);
+  assert(runs_finished <=
+         num_running_compaction_sorted_runs_->load(std::memory_order_relaxed));
+  num_running_compaction_sorted_runs_->fetch_sub(runs_finished,
+                                                 std::memory_order_relaxed);
+}
+
+size_t DBStatsCallback::GetNumberCompactionSortedRuns(const Compaction* c) {
+  assert(c);
+  if (c->IsTrivialMove() || c->deletion_compaction()) {
+    return 0;
+  }
+  // Return the number of L0 files + number of non-L0 levels
+  if (c->start_level() == 0) {
+    assert(0 < c->num_input_levels());
+    assert(c->level(0) == 0);
+    size_t num_l0_files = c->num_input_files(0);
+    size_t num_non_l0_levels = c->num_input_levels() - 1;
+    return num_l0_files + num_non_l0_levels;
+  }
+  return c->num_input_levels();
+}
+
```

Comment:
I briefly remembered you mentioned counting the input L0 files and non-L0 level doesn't work for  c->IsTrivialMove() || c->deletion_compaction(). Did you decided to not counting for trivial move and deletion compaction then?

Reply:
There is this check at the top
```cpp
  if (c->IsTrivialMove() || c->deletion_compaction()) {
    return 0;
  }
```

(1 or 2 PR variations back we decided not to count trivial moves and deletion compactions)